### ISSUE TITLE
💅 useExportType doesn't handle `export { T } from './file2'`

### ISSUE THREAD

**Issue by Akuukis** [biomejs/biome#7293]
### Environment information

```bash
See playground:

// As expected (noUnresolvedImports): "All these imports are only used as types."
import { T } from './file2'

// As expected (useExportType): "All exports are only types."
type T2 = T
export { T2 }

// Unexpected (useExportType?): I expect some rule to tell me that below should be this instead:
//     export type { T } from './file2'
export { T } from './file2'

// Usecase: Typescript doesn't complain, babel won't strip it, and then in webpack HMR it errors with:
// "export 'T' (reexported as 'T') was not found in './file2' (possible exports: default)"
```

### Rule name

useExportType

### Playground link

https://biomejs.dev/playground/?code=LwAvACAAQQBzACAAZQB4AHAAZQBjAHQAZQBkACAAKABuAG8AVQBuAHIAZQBzAG8AbAB2AGUAZABJAG0AcABvAHIAdABzACkAOgAgACIAQQBsAGwAIAB0AGgAZQBzAGUAIABpAG0AcABvAHIAdABzACAAYQByAGUAIABvAG4AbAB5ACAAdQBzAGUAZAAgAGEAcwAgAHQAeQBwAGUAcwAuACIACgBpAG0AcABvAHIAdAAgAHsAIABNAHkAVAB5AHAAZQAgAH0AIABmAHIAbwBtACAAJwAuAC8AbwB0AGgAZQByAEYAaQBsAGUAJwAKAAoALwAvACAAQQBzACAAZQB4AHAAZQBjAHQAZQBkACAAKAB1AHMAZQBFAHgAcABvAHIAdABUAHkAcABlACkAOgAgACIAQQBsAGwAIABlAHgAcABvAHIAdABzACAAYQByAGUAIABvAG4AbAB5ACAAdAB5AHAAZQBzAC4AIgAKAHQAeQBwAGUAIABNAHkAVAB5AHAAZQBSAGUAZABlAGYAaQBuAGUAZAAgAD0AIABNAHkAVAB5AHAAZQAKAGUAeABwAG8AcgB0ACAAewAgAE0AeQBUAHkAcABlAFIAZQBkAGUAZgBpAG4AZQBkACAAfQAKAAoALwAvACAAVQBuAGUAeABwAGUAYwB0AGUAZAAgACgAPwA%2FAD8AKQA6ACAASQAgAGUAeABwAGUAYwB0ACAAcwBvAG0AZQAgAHIAdQBsAGUAIAB0AG8AIAB0AGUAbABsACAAbQBlACAAdABoAGEAdAAgAGIAZQBsAG8AdwAgAHMAaABvAHUAbABkACAAYgBlACAAdABoAGkAcwAgAGkAbgBzAHQAZQBhAGQAOgAKAC8ALwAgACAAIAAgACAAZQB4AHAAbwByAHQAIAB0AHkAcABlACAAewAgAE0AeQBUAHkAcABlACAAfQAgAGYAcgBvAG0AIAAnAC4ALwBvAHQAaABlAHIARgBpAGwAZQAnAAoAZQB4AHAAbwByAHQAIAB7ACAATQB5AFQAeQBwAGUAIAB9ACAAZgByAG8AbQAgACcALgAvAG8AdABoAGUAcgBGAGkAbABlACcACgAKAC8ALwAgAFUAcwBlAGMAYQBzAGUAOgAgAFQAeQBwAGUAcwBjAHIAaQBwAHQAIABkAG8AZQBzAG4AJwB0ACAAYwBvAG0AcABsAGEAaQBuACwAIABiAGEAYgBlAGwAIAB3AG8AbgAnAHQAIABzAHQAcgBpAHAAIABpAHQALAAgAGEAbgBkACAAdABoAGUAbgAgAGkAbgAgAHcAZQBiAHAAYQBjAGsAIABIAE0AUgAgAGkAdAAgAGUAcgByAG8AcgBzACAAdwBpAHQAaAA6AAoALwAvACAAIgBlAHgAcABvAHIAdAAgACcATQB5AFQAeQBwAGUAJwAgACgAcgBlAGUAeABwAG8AcgB0AGUAZAAgAGEAcwAgACcATQB5AFQAeQBwAGUAJwApACAAdwBhAHMAIABuAG8AdAAgAGYAbwB1AG4AZAAgAGkAbgAgACcALgAvAG8AdABoAGUAcgBGAGkAbABlACcAIAAoAHAAbwBzAHMAaQBiAGwAZQAgAGUAeABwAG8AcgB0AHMAOgAgAGQAZQBmAGEAdQBsAHQAKQAiAAoA&ruleDomains.project=all

### Expected result

I expected `useExportType` to handle exports from other files.

Context: My codebase had a few redefine-and-reexport statements, that I refactored into one-line export statements from the other file and missed the `type` keyword. This caught me off-guard, as I relied on Biome under understanding that `useExportType` will handle exports in any export syntax form, because the description of the rule suggests so by talking about `export` and `type` keyword in general. I'm confused what to make of it and filing as a bug, but depending on your stance this report can instead be either:
- bug: `useExportType` should handle re-exports
- documentation: clarify that `useExportType` handles only exports from the same file, not exports from other files
- feature request: add a new `useReexportType` (or a similar name)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
